core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -151,6 +155,7 @@ public FuzzyQueryBuilder(String fieldName, boolean value) {
      * @param value The value of the term
      */
     public FuzzyQueryBuilder(String fieldName, Object value) {
+        DEPRECATION_LOGGER.deprecated("{} query is deprecated. Instead use the [match] query with fuzziness parameter", NAME);
```

Comment:
@javanna I looked into adding rest tests that check the warnings like @nik9000 suggested, I could add a test similar to the one that the docs-tests generate but there seems to be no other rest test for "fuzzy" at the moment and it feels strange to introduce tests for a deprecated query. As mentioned above, the docs tests already check the warning headers. Are you okay with leaving it like this?